engine/table/src/test/java/io/deephaven/engine/updategraph/impl/TestEventDrivenUpdateGraph.java

```diff
@@ -165,6 +167,40 @@ public void testSimpleModify() {
         }
     }
 
+    @Test
+    public void testRefreshRace() throws ExecutionException, InterruptedException, TimeoutException {
+        final EventDrivenUpdateGraph eventDrivenUpdateGraph = EventDrivenUpdateGraph.newBuilder("TestEDUG").build();
+
+        final MutableInt sourceRefreshCount = new MutableInt(0);
+        final Runnable sleepingSource = () -> {
+            try {
+                Thread.sleep(100);
+                sourceRefreshCount.increment();
+            } catch (InterruptedException e) {
+                Assert.fail("Interrupted while sleeping");
+            }
+        };
+        eventDrivenUpdateGraph.addSource(sleepingSource);
+
+        final int numConcurrentRefreshes = 10;
+        final Future<?>[] refreshFutures = new Future[numConcurrentRefreshes];
+        final ExecutorService executor = Executors.newFixedThreadPool(numConcurrentRefreshes);
+        try {
+            for (int cri = 0; cri < numConcurrentRefreshes; ++cri) {
+                refreshFutures[cri] = executor.submit(eventDrivenUpdateGraph::requestRefresh);
+                Thread.sleep(10);
+            }
+            for (final Future<?> refreshFuture : refreshFutures) {
+                refreshFuture.get();// 1, TimeUnit.SECONDS);
```

Comment:
why no time out?  Seems like it would be beneficial to prevent a hung test when something goes horribly wrong.

Reply:
I didn't intend to merge it like this, was part of my efforts to prove that the test failed with the old code. Fixed now.